openssl/src/x509/mod.rs

```diff
@@ -1452,8 +1457,13 @@ impl GeneralNameRef {
                 return None;
             }
 
-            let ptr = ASN1_STRING_get0_data((*self.as_ptr()).d as *mut _);
-            let len = ffi::ASN1_STRING_length((*self.as_ptr()).d as *mut _);
+            #[cfg(boringssl)]
+            let d: *const ffi::ASN1_STRING = std::mem::transmute((*self.as_ptr()).d);
```

Comment:
Why is this transmuting into a raw pointer type?

Reply:
It was transmuting because there's a union there, not a pointer. In your handwritten bindings you've pretended it is a void*, then cast it to an ASN1_STRING, and blocklist it from the bindgen.

The transmute still wasn't the correct call, as there's a union member that gives you raw pointer access. I've switched to use that instead in the boringssl side of the ifdef, and you could consider using it if you want to stop blocklisting GENERAL_NAME in your bindgen openssl path.